buildtest/config.py

```diff
@@ -79,5 +74,7 @@ def load_configuration(config_path=None):
     return load_schema(config_path)
 
 
-# Run on init, so we only load once
-config_opts = load_configuration()
```

Comment:
I'm worried about this line - we are removing the ability to import config_opts from anywhere (you don't think this is important - to load them at init onset and then not need to do it again?) It would mean that there could be a race condition of loading some config opts, having it changed, and then re-loading all during a run.

That's my only concern with this change - let me know your thoughts.

Reply:
I observed this change when i was trying to implement this feature and noticed that ``config_opts`` was only loading default configuration (``~/.buildtest/settings.yml``) and it was only imported in two different files

https://github.com/HPC-buildtest/buildtest-framework/blob/74735173b772e924eb7504c01a7e70550ca3cb76/buildtest/menu/config.py#L3

https://github.com/HPC-buildtest/buildtest-framework/blob/74735173b772e924eb7504c01a7e70550ca3cb76/buildtest/menu/build.py#L17

So i figured that if we need to retrieve ``config_opts`` instead of loading this value we can import and call method ``get_default_configuration`` https://github.com/shahzebsiddiqui/buildtest-framework/blob/issue/258/buildtest/config.py#L77 which would be equivalent to using config_opts. 

The issue was having to either preserve ``config_opts`` as a variable and have this passed to other methods while supporting a new configuration file which needs to load the configuration and get an updated ``config_opts`` that can be passed to ``BuildExecutor`` class

https://github.com/shahzebsiddiqui/buildtest-framework/blob/0c5a260dc338417281b6cc57e5d7f1df71e7e6a4/buildtest/menu/build.py#L99

I was also thinking that checking configuration should be done at entry point of buildtest build as pose to before calling the parser. This was to support the command line option and not have to do load configuration again. That's why i removed line https://github.com/HPC-buildtest/buildtest-framework/pull/261/files#diff-b01bc590c712803d7252b15aa8dbd8bcL18

Reply:
okay, I'm not sure that I totally am totally comfortable with this change, but I think you've put a good amount of thought into it and I don't think it hurts too badly to load on demand.

Reply:
I have ran into this cyclic import issue a while back about time when i was refactoring code about time when you joined. I did like the ability to load configuration once and pass ``config_opts`` as variable to other modules, but right now it not being used much so didnt make sense to keep it and find way to add this feature. If codebase changes where ``config_opts`` is needed by several python modules, i think then we should seriously consider defining as a global variable and call it once. It's not good to call ``get_default_configuration`` or ``load_configuration`` several times as it reading file every-time. 

Also note i call ``load_configuration`` and ``check_configuration`` https://github.com/HPC-buildtest/buildtest-framework/pull/261/files#diff-dd4fa250655dc41a1197ad6dc7b8a126R82-R85 

Even though check_configuration calls load_configuration, I wanted these to be separate methods. This way check_configuration is done once while load_configuration can be called one or more times as needed to retrieve the configuration as a dict.